### ISSUE TITLE
False-positive RET504 "Unnecessary variable assignment before `return` statement"

### ISSUE THREAD

**Issue by actionless** [charliermarsh/ruff#1290]
ruff 0.0.187

```python
import sys
from time import time


def func() -> float:
    result = time()
    sys.stdout.write('log smth\n')
    return result
```

```console
$ ruff test_ruff_ret_504.py
Found 1 error(s).
test_ruff_ret_504.py:8:12: RET504 Unnecessary variable assignment before `return` statement
```

**Comment by squiddy** [comment]
See also https://github.com/charliermarsh/ruff/issues/1233, which was closed recently.

> I'm torn on this! Almost any statement can be side-effectful -- even something like a = b + c could actually involve operator overloading under-the-hood and thus make use of bar. So, I guess the suggestion is that we make this check much more conservative, and only flag when there are no expressions between the assignment and the return?

Which is not to say that this report/request is invalid, the question on how to deal with that sensibly is still open.

**Comment by actionless** [comment]
i think it's reasonable to assume that any statement is side-effectful - at least because it takes some time to execute

**Comment by charliermarsh** [comment]
Alright, maybe we just change this to only allow obviously non-effectful statements (like `pass`) between assignment and return.

**Comment by charliermarsh** [comment]
Going out in [v0.0.189](https://github.com/charliermarsh/ruff/releases/tag/v0.0.189).
